tests/helpers/kubernetes/utils.py

```diff
@@ -62,45 +52,66 @@ def run_k8s_monitors_test(  # pylint: disable=too-many-locals,too-many-arguments
                                              agent container logs
     test_timeout (int):                    Timeout in seconds to wait for metrics/dimensions
     """
-    if yamls is None:
-        yamls = []
-    if expected_dims is not None:
-        if observer:
-            expected_dims = expected_dims.union(get_observer_dims_from_selfdescribe(observer))
-        expected_dims = expected_dims.union({"kubernetes_cluster"})
-
-    with run_k8s_with_agent(agent_image, minikube, monitors, observer, namespace, yamls, yamls_timeout) as [
-        backend,
-        agent,
-    ]:
+    args = dict(
```

Comment:
I don't like this at all.  If the whole point is to try and get around the too-many-arguments rule then just disable it and let it be.

Reply:
Ok, I'll revert.